lib/nrf91_lwm2m_app/Kconfig

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig NRF91_LWM2M_APP
+	bool "nRF91 LWM2M client application library"
+	select NRF_MODEM_LIB
+	select LWM2M
+	select NETWORKING
+	select NET_SOCKETS
+	select NET_SOCKETS_OFFLOAD
+
+if NRF91_LWM2M_APP
+
+config NET_NATIVE
+	bool "Use OS native network stack(NOT SUPPORTED)"
+	default n
+
+config LWM2M_DNS_SUPPORT
+	bool "DNS support"
+	default y
+
+config LWM2M_DTLS_SUPPORT
+	bool "DTLS support"
+	default y
+
+config LWM2M_PEER_PORT
+	int "Peer port"
+	default 5684
+
+config LWM2M_SECURITY_KEY_SIZE
+	int "Security key size. Support HEX style PSK values (double the size + NULL char)"
+	default 33
+
+menu "CoAP adjustments"
+
+config COAP_INIT_ACK_TIMEOUT_MS
+	int "Extended CoAP retry timing to for LTE-M/NB-IoT [ms]"
+	default 10000
+
+config COAP_EXTENDED_OPTIONS_LEN
+	bool "Enable CoAP extended option length"
+	default y
+
+if COAP_EXTENDED_OPTIONS_LEN
+config COAP_EXTENDED_OPTIONS_LEN_VALUE
+	int "CoAP extended option length"
+	default 40
+
+endif #COAP_EXTENDED_OPTIONS_LEN
+endmenu
+
+config LWM2M_SERVER_DEFAULT_PMIN
```

Comment:
Actually, I don't think that configs like these should be altered by the library in any way. The `pmin`/`pmax` for instance is a core LwM2M feature and I don't think the library needs to alter this value in any way. Same goes with some other configs like JSON support or COAP ACK timeout.

Reply:
Ok, will fix that one. I was also thinking should I bring any of the changes on the overlay files as part of Kconfig. I mean here the bootstrap server settings, queue mode and NB-IoT-specific settings.

Reply:
I'd be really cautious with moving these configs to be honest - the library should provide the default implementation of certain objects, not enforce specific configurations. For instance the port number - it's tightly bound with the sever being used, I don't think the library should enforce any value - I'd say it's an application-specifc config (like the server URL).

In my opinion we shouldn't introduce too many configs to the library. I was rather thinking we should have configs like `CONFIG_NRF91_LWM2M_APP_FIRMWARE_UPDATE_OBJ` that would enable the Firmware Object implementation in the library, and which could `select` any other configs from the core library to make it work. The user would then have the flexibility to disable parts of the library that he wants to re-implement differently.

Reply:
>  I was rather thinking we should have configs like CONFIG_NRF91_LWM2M_APP_FIRMWARE_UPDATE_OBJ that would enable the Firmware Object implementation in the library, and which could select any other configs from the core library to make it work.

That is still to be done but I hope the split of the configurations is now you hope it to be.

Reply:
Hopefully fixed fully now.